core/src/test/java/io/github/tcdl/msb/api/ChannelMonitorIT.java

```diff
@@ -37,10 +58,158 @@ public void testAnnouncement() throws InterruptedException {
             announcementReceived.countDown();
         };
 
-        DefaultChannelMonitorAggregator channelMonitorAggregator = new DefaultChannelMonitorAggregator(msbContext, new ScheduledThreadPoolExecutor(1), handler);
+        channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
         channelMonitorAggregator.start(false, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_TIMEOUT_MS);
 
         channelMonitorAgent.producerTopicCreated(TOPIC_NAME);
         assertTrue("Announcement was not received", announcementReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
     }
+
+    @Test
+    public void testAnnouncementUnexpectedMessage() throws InterruptedException {
+        String TOPIC_NAME = "topic1";
+        ChannelMonitorAgent channelMonitorAgent = new DefaultChannelMonitorAgent(msbContext);
+
+        CountDownLatch announcementReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertTrue(stats.getTopicInfoMap().containsKey(TOPIC_NAME));
+            assertEquals(1, stats.getTopicInfoMap().get(TOPIC_NAME).getProducers().size());
+            announcementReceived.countDown();
+        };
+
+        //simulate unexpectedMessage was consumed from in TOPIC_ANNOUNCE namespace
+        MockAdapter.pushRequestMessage(Utils.TOPIC_ANNOUNCE,
+                Utils.toJson(TestUtils.createMsbRequestMessageWithSimplePayload(Utils.TOPIC_ANNOUNCE), msbContext.getPayloadMapper()));
+
+        channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(false, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_TIMEOUT_MS);
+        channelMonitorAgent.producerTopicCreated(TOPIC_NAME);
+
+        assertTrue("Announcement was not received", announcementReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME * 2, TimeUnit.MILLISECONDS));
```

Comment:
Test is 100% valid.
no way, we are waiting not for Response but End event..but yes in some way if you broke  our end event logic this will pass. But this is all i can do here